test/python/quantum_info/states/test_statevector.py

```diff
@@ -335,6 +344,20 @@ def test_tensor(self):
             self.assertEqual(state.dims(), (3, 2))
             assert_allclose(state.data, target)
 
+    def test_inner(self):
+        """Test inner method."""
+        for _ in range(10):
+            seed = np.random.randint(0, np.iinfo(np.int32).max)
+            rng = np.random.default_rng(seed)
+            dim = 2 ** rng.integers(1, 4)
+            vec0 = Statevector(self.rand_vec(dim))
+            vec1 = Statevector(self.rand_vec(dim))
+            target = 0
+            for i in range(vec0.dim):
+                target += np.conj(vec0.data[i]) * vec1.data[i]
```

Comment:
This is equivalent to `np.vdot(vec0.data, vec1.data)` (which would also be significantly faster).

Reply:
Thanks.
```suggestion
            target = np.vdot(vec0.data, vec1.data)
```